profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/NicApiLiveTest.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.jclouds.profitbricks.rest.domain.Server;
+import org.apache.jclouds.profitbricks.rest.ids.ServerRef;
+import org.apache.jclouds.profitbricks.rest.internal.BaseProfitBricksLiveTest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "live", testName = "NicApiLiveTest")
+public class NicApiLiveTest extends BaseProfitBricksLiveTest {
+   
+   DataCenter dataCenter;
+   Server testServer;
+   Nic testNic;
+  
+   @BeforeClass
+   public void setupTest() {
+      dataCenter = createDataCenter();
+      assertDataCenterAvailable(dataCenter);
+            
+      testServer = api.serverApi().createServer(
+         Server.Request.creatingBuilder()
+            .dataCenterId(dataCenter.id())
+            .name("jclouds-node")
+            .cores(1)
+            .ram(1024)
+            .build());
+      
+      assertNodeAvailable(ServerRef.create(dataCenter.id(), testServer.id()));
+   }
+   
+   @AfterClass(alwaysRun = true)
+   public void teardownTest() {
+      if (dataCenter != null)
+         deleteDataCenter(dataCenter.id());
+   }
+     
+   @Test
+   public void testCreateNic() {
+      assertNotNull(dataCenter);
+            
+      testNic = nicApi().createNic(
+              Nic.Request.creatingBuilder()
+              .dataCenterId(dataCenter.id())
+              .serverId(testServer.id())
+              .name("jclouds-nic")
+              .lan(1)
+              .build());
+
+      assertNotNull(testNic);
+      assertEquals(testNic.properties().name(), "jclouds-nic");
+      assertNicAvailable(testNic);
+   }
+   
+
+   @Test(dependsOnMethods = "testCreateNic")
+   public void testGetNic() {
+      Nic nic = nicApi().getNic(dataCenter.id(), testServer.id(), testNic.id());
+
+      assertNotNull(nic);
+      assertEquals(nic.id(), testNic.id());
+   }
+
+   @Test(dependsOnMethods = "testCreateNic")
+   public void testList() {
+      List<Nic> nics = nicApi().getList(dataCenter.id(), testServer.id());
+
+      assertNotNull(nics);
+      assertFalse(nics.isEmpty());
+      assertEquals(nics.size(), 1);
+   }
+   
+   @Test(dependsOnMethods = "testGetNic")
+   public void testUpdateNic() {
+      assertDataCenterAvailable(dataCenter);
+      
+      nicApi().updateNic(
+              Nic.Request.updatingBuilder()
+              .dataCenterId(testNic.dataCenterId())
+              .serverId(testServer.id())
+              .id(testNic.id())
+              .name("apache-nic")
+              .build());
+
+      assertNicAvailable(testNic);
+      
+      Nic nic = nicApi().getNic(dataCenter.id(), testServer.id(), testNic.id());
+      
+      assertEquals(nic.properties().name(), "apache-nic");
+   }
+   
+
+   @Test(dependsOnMethods = "testUpdateNic")
+   public void testDeleteNic() {
```

Comment:
But if the update test fails, we don't consider the tests passing? Also I'd still have to wait since I don't really want to update and delete at the same time.